### PR TITLE
Feature/nec 11/granularity search tests and minor improvements

### PR DESCRIPTION
**Pull request by alexeyvy**
The `UnionSearchService` is now covered with a few unit tests. Also, some minor improvements covering edge cases done.

### CODE REVIEW

File: models/classes/search/aggregator/UnionSearchService.php [hunk 1]
```diff
@@ -26,47 +26,74 @@
...
 /**
  * Class UnionSearchService
+ *
+ * @todo implementing some Search interface was considered, however, the actual one cannot be
+ * the one because it currently violates the 'Interface Segregation Principle' by defining SERVICE_ID and including a lot of functions
+ * which really makes it SearchService Interface instead of Search Interface
+ *
  * @package oat\tao\model\search\aggregator
  */
-class UnionSearchService extends ConfigurableService implements UnionSearchInterface
```
Comment (tikhanovichA) [inline models/classes/search/aggregator/UnionSearchService.php]: Do we need to delete UnionSearchInterface?

Comment (tikhanovichA) [reply models/classes/search/aggregator/UnionSearchService.php]: If yes, what about SERVICE_ID?

File: models/classes/search/aggregator/UnionSearchService.php [hunk 2]
```diff
@@ -26,47 +26,74 @@
...
+class UnionSearchService extends ConfigurableService
```
Comment (alexeyvy) [inline models/classes/search/aggregator/UnionSearchService.php]:
```suggestion
class UnionSearchService extends ConfigurableService implements UnionSearchInterface
```

File: models/classes/search/aggregator/UnionSearchService.php (hunk 1 above)
Comment (alexeyvy) [reply models/classes/search/aggregator/UnionSearchService.php]: fixed
